worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,205 @@
+        for item_name, item_data in self.item_table.items():
+            # These item types are handled individually
+            if item_data.item_type in [CivVICheckType.PROGRESSIVE_DISTRICT, CivVICheckType.ERA, CivVICheckType.GOODY]:
+                continue
+
+            # If we're using progressive districts, we need to check if we need to create a different item instead
+            item_to_create = item_name
+            item: CivVIItemData = self.item_table[item_name]
+            if self.options.progression_style != "none":
+                if item.progression_name:
+                    item_to_create = self.item_table[item.progression_name].name
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+            if item.civ_name in early_items:
+                self.multiworld.early_items[self.player][item_to_create] = 1
+            elif self.item_table[item_name].era in [EraType.ERA_ATOMIC, EraType.ERA_INFORMATION, EraType.ERA_FUTURE]:
+                for location in early_locations:
+                    found_location = None
+                    try:
+                        found_location = self.get_location(location.name)
+                        forbid_item(found_location, item_to_create, self.player)
+                    except KeyError:
+                        pass
+
+        # Era items
+        if self.options.progression_style == "eras_and_districts":
+            # Add one less than the total number of eras (start in ancient, don't need to find it)
+            for era in EraType:
+                if era.value == "ERA_ANCIENT":
+                    continue
+                progressive_era_item = self.item_table.get("Progressive Era")
+                if progressive_era_item:
+                    self.multiworld.itempool += [self.create_item(progressive_era_item.name)]
+                else:
+                    logging.error("Progressive Era item not found in item_table.")
+            self.multiworld.early_items[self.player]["Progressive Era"] = 2
+
+        num_filler_items = 0
+        # Goody items, create 10 by default if options are enabled
+        if self.options.shuffle_goody_hut_rewards:
+            num_filler_items += 10
+
+        if self.options.boostsanity:
+            num_filler_items += len(get_boosts_data())
+
+        filler_count = {rarity: math.ceil(FILLER_DISTRIBUTION[rarity] * num_filler_items) for rarity in FillerItemRarity.__reversed__()}
+        min_count = 1
+        # Add filler items by rarity
+        total_created = 0
+        for rarity, count in filler_count.items():
+            for _ in range(max(min_count, count)):
+                if total_created >= num_filler_items:
+                    break
+                self.multiworld.itempool += [self.create_item(
+                    get_random_filler_by_rarity(self, rarity).name)]
+                total_created += 1

```

Comment:
Have you considered doing something like this instead?
```suggestion
        filler_count = {rarity: math.ceil(FILLER_DISTRIBUTION[rarity] * num_filler_items) for rarity in FillerItemRarity.__reversed__()}
        filler_count[FillerItemRarity.COMMON] -= sum(filler_count.values()) - num_filler_items
        self.multiworld.itempool += [
            self.create_item(get_random_filler_by_rarity(self, rarity).name)
            for rarity, count in filler_count.items()
            for _ in range(count)
        ]
```

Reply:
Nice, much more elegant 🔥